src/entt/entity/storage.hpp

```diff
@@ -233,25 +307,41 @@ template<typename... Lhs, typename... Rhs>
  * @tparam Allocator Type of allocator used to manage memory and elements.
  */
 template<typename Type, typename Entity, typename Allocator, typename>
-class basic_storage: public basic_sparse_set<Entity, typename std::allocator_traits<Allocator>::template rebind_alloc<Entity>> {
+class basic_storage
+    : public basic_sparse_set<Entity, typename std::allocator_traits<Allocator>::template rebind_alloc<Entity>>,
+      protected Allocator,
```

Comment:
Inheriting from an allocator and a traits class sounds soooooo bad.
Why aren't they just members? I don't see any benefit from inheritance. here.

Reply:
This is to enable empty type optimization, I can changed it as mentioned in one of my initial comments above.

Reply:
I specifically implemented and used the `compressed_pair` class for the same purpose and the main goal was to exploit ETO without having to inherit from an allocator. 👍🙂